mnemosyne/diagnose.py

```
@@ -62,6 +62,85 @@ def _safe_env(name: str) -> str:
+    tables = {
+        row[0]
+        for row in conn.execute(
+            "SELECT name FROM sqlite_master WHERE type='table'"
+        ).fetchall()
+    }
+    live_id_tables = [
+        table
+        for table in ("working_memory", "memories", "episodic_memory")
+        if table in tables
+    ]
+    live_ids_sql = " UNION ".join(f"SELECT id FROM {table}" for table in live_id_tables)
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)